Automattic/jetmon
# Stats: Replace hot-shots library with raw socket sends

The Systems team requested that we not rely on the `hot-shots` library to reduce the number of external dependencies. This PR replaces the use of the external library with a custom set of functions that use node's built-in `dgram` module. The new functions are written to be a drop-in replacement for how the existing code made calls to the `hot-shots` library.

Note: I changed the `logger` var definition in `httpcheck.js` to global so that calls `statsdClient` can use `logger` to record errors when called from a worker thread.

### Testing
- Apply the PR.
- Run Jetmon.
- Wait a few minutes for stats to be sent.
- Verify that stats still populate in Graphite (127.0.0.1:8088). The following are good metrics to check:
  - For increment, verify that `stats_counts.com.jetpack.jetmon.jetmon.docker.stats.workers.free.count` is updating properly.
  - For timing, verify that `stats.timers.com.jetpack.jetmon.jetmon.docker.db.get_next_batch.count` is updating properly.
